### PR TITLE
Add carriage return to log message when using --tty flag

### LINKED ISSUE #961: podman log messages do not print carriage return when using --tty flag
<!--
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
> /kind bug

**Description**

When using podman to run a container with the --tty flag, the log messages printed with the --log-level option do not include the carriage return.

**Steps to reproduce the issue:**

```podman --log-level info run --tty --rm hello-world```


**Describe the results you received:**
The following output:
```
INFO[0000] [graphdriver] using prior storage driver: overlay 
INFO[0000] CNI network podman (type=bridge) is used from /etc/cni/net.d/87-podman-bridge.conflist 
INFO[0000] Initial CNI setting succeeded                
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                                                                                                                                                                                                                            INFO[0000] About to add CNI network cni-loopback (type=loopback) 
                                                INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                               INFO[0000] About to add CNI network podman (type=bridge) 
                                                                                        WARN[0001] failed to parse language "en_US.utf8": language: tag is not well-formed 

Hello from Docker!
This message shows that your installation appears to be working correctly.
```

**Describe the results you expected:**
The following output:
```
INFO[0000] [graphdriver] using prior storage driver: overlay 
INFO[0000] CNI network podman (type=bridge) is used from /etc/cni/net.d/87-podman-bridge.conflist 
INFO[0000] Initial CNI setting succeeded                
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                                                                                                                                                                                                                            INFO[0000] About to add CNI network cni-loopback (type=loopback) 
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
INFO[0000] About to add CNI network podman (type=bridge) 
WARN[0001] failed to parse language "en_US.utf8": language: tag is not well-formed 

Hello from Docker!
This message shows that your installation appears to be working correctly.
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:       0.6.3
Go Version:    go1.10.3
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  MemFree: 4760469504
  MemTotal: 16700174336
  SwapFree: 16868438016
  SwapTotal: 16869486592
  arch: amd64
  cpus: 4
  hostname: wildfox
  kernel: 4.17.1-1-default
  os: linux
  uptime: 19h 49m 25.02s (Approximately 0.79 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
store:
  ContainerStore:
    number: 4
  GraphDriverName: overlay
  GraphOptions: []
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 5
  RunRoot: /var/run/containers/storage
```

### PR DESCRIPTION
**Pull request by marcov**
Add a raw text formatter for logrus to be used when terminal is in raw
mode (i.e. when allocating a pseudo-TTY for the container). The raw text
formatter terminates the log messages with the corret \r\n sequence.

Fixes #961

### CODE REVIEW

**Comment by mheon** [comment]
I think we're actually terminating with `\n\r` with this, but if it works and the code is simpler I'm absolutely fine with that. LGTM.

File: cmd/podman/utils.go [hunk 1]
```diff
@@ -171,3 +176,19 @@ func resizeTty(resize chan remotecommand.TerminalSize, resizeTerminate chan inte
...
+	logrus.SetFormatter(&logrus.TextFormatter{})
+	return term.RestoreTerminal(os.Stdin.Fd(), state)
+}
+
+func (f *RawTtyFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	textFormatter := logrus.TextFormatter{}
+	bytes, err := textFormatter.Format(entry)
+
+	if err == nil {
+		bytes = append(bytes, '\r')
```
Comment (rhatdan) [inline cmd/podman/utils.go]: Is '\r' enough or does it need `\n`

**Comment by marcov** [comment]
@mheon: you are right. In my tests ```\n\r``` is visually equivalent to the canonical ```\r\n``` representation. Swapping the two bytes is matter of:
```diff
       if err == nil {
-               bytes = append(bytes, '\r')
+               bytes[len(bytes)-1] = '\r'
+               bytes = append(bytes, '\n')
        }

```

**Comment by marcov** [comment]
@rhatdan ```\n``` is already included by the logrus text formatter as the last byte in the buffer. I am appending an extra ```\r``` to it.

File: cmd/podman/utils.go (hunk 1 above)
Comment (marcov) [reply cmd/podman/utils.go]: ```\n``` is already included by the logrus text formatter as the last byte in the buffer. I am appending an extra ```\r``` to it.

**Comment by rhatdan** [comment]
I just tried this locally and it looks awesome
@marcov Thanks
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**